docs/modules/ROOT/pages/quarkus-azure-servicebus.adoc

```diff
@@ -0,0 +1,248 @@
+
+== Installation
+
+Add a dependency on `io.quarkiverse.azureservices:quarkus-azure-servicebus`.
+
+For instance, with Maven, add the following dependency to your POM file:
+
+[source,xml,subs=attributes+]
+----
+<dependency>
+    <groupId>io.quarkiverse.azureservices</groupId>
+    <artifactId>quarkus-azure-servicebus</artifactId>
+    <version>{project-version}</version>
+</dependency>
+----
+
+== How to Use It
+
+Once you have added the extension to your project, follow the next steps, so you can inject `com.azure.messaging.servicebus.ServiceBusClientBuilder` object in your application to create clients for producing and consuming messages from Azure Service Bus.
+
+=== Setup your Azure Environment
+
+include::includes/azure-setup-environment.adoc[]
+
+Create an Azure resource group with the az group create command.
+A resource group is a logical container into which Azure resources are deployed and managed.
+
+[source,shell]
+----
+az group create \
+    --name rg-quarkus-azure-servicebus \
+    --location westus
+----
+
+Create an Azure Service Bus namespace and a queue with the following commands:
+
+[source,shell]
+----
+az servicebus namespace create \
+    --name sbnamespace20250604 \
+    --resource-group rg-quarkus-azure-servicebus
+
+az servicebus queue create \
+    --name test-queue \
+    --namespace-name sbnamespace20250604 \
+    --resource-group rg-quarkus-azure-servicebus
+----
+
+If you log into the http://portal.azure.com/[Azure portal], you can see the resource group and the Azure Service Bus namespace you created.
+
+=== Configure the Azure Service Bus Client Builder
+
+As you can see below in the _Extension Configuration Reference_ section, either the property `quarkus.azure.servicebus.namespace` or the property `quarkus.azure.servicebus.connection-string` is required if the extension is enabled.
+
+You have two options to authenticate to Azure Service Bus, either with Microsoft Entra ID or connection string. The following sections describe how to authenticate with both options. For optimal security, it is recommended to use Microsoft Entra ID for authentication.
+
+==== Authenticating to Azure Service Bus with Microsoft Entra ID
+
+You can authenticate to Azure Service Bus with Microsoft Entra ID. Run the following commands to assign the `Azure Service Bus Data Owner` role to the signed-in user as a Microsoft Entra identity.
+
+[source,shell]
+----
+SERVICEBUS_RESOURCE_ID=$(az servicebus namespace show \
+    --resource-group rg-quarkus-azure-servicebus \
+    --name sbnamespace20250604 \
+    --query 'id' \
+    --output tsv)
+OBJECT_ID=$(az ad signed-in-user show --query id -o tsv)
+
+az role assignment create \
+    --role "Azure Service Bus Data Owner" \
+    --assignee ${OBJECT_ID} \
+    --scope $SERVICEBUS_RESOURCE_ID
+----
```

Comment:
This command fails for me. I get
```shell
$ az role assignment create \
    --role "Azure Service Bus Data Owner" \
    --assignee ${OBJECT_ID} \
    --scope $SERVICEBUS_RESOURCE_ID
'. If the assignee is an appId, make sure the corresponding service principal is created with 'az ad sp create --id <id redacted>
```
I had to manually create the role assignment.

Can you please verify whether this is a general issue?

Reply:
Since the doc just describes how to run the sample app locally, the type of expected assignee for this command is a user instead of application (service principal, managed identity, etc.):
```
OBJECT_ID=$(az ad signed-in-user show --query id -o tsv)
```

Are you assigning the role to the user or managed identity? If you're trying to assign the role to a user-assignmed managed identity, you should retrieve the `OBJECT_ID` as below:
```
OBJECT_ID=$(az identity show \
    --resource-group "${RESOURCE_GROUP_NAME}" \
    --name "${USER_ASSIGNED_IDENTITY_NAME}" \
    --query 'principalId' \
    --output tsv)
```